### DISCUSSION TITLE
Using storage policies without Supabase Auth

### DISCUSSION THREAD

**Discussion by nightspite** [supabase/supabase#12274] [Questions]
I created a small app with t3-stack connected to a Supabase Database. I use Next Auth for the auth, user data is stored in the database provided by Supabase. 

The question:
Is it possible to use this custom User entity to create policies and permission for the Supabase Storage? and if so how can it be done?

There are two use cases for me:
- uploading avatar images -> no auth needed, public access
- uploading user files that could only be shared between two users -> auth needed, private access

**Comment by jlia0** [comment]
Asking for the same^^^
